### ISSUE TITLE
Feature request: Add support for ClangCl on MSVC

### ISSUE THREAD

**Issue by emmenlau** [oneapi-src/oneDAL#2115]
**Describe the bug**
It would be great if oneDAL could be compiled with the clang-cl compiler frontend for Visual Studio. I have tried to use ClangCl but there seem to be some warnings or errors that make the build fail. Since ClangCl is very similar to clang but also to Visual Studio, it seems it is well aligned with the already supported compilers. And on Windows it generates often much faster code than Visual Studio alone.

**Comment by emmenlau** [comment]
@napetrov yes correct, I'd like to build the library with clang-cl from LLVM, as a frontend to Visual Studio.

**Comment by napetrov** [comment]
@emmenlau  - the way how some optimizations are enabled is based on use of compiler intrinsics and some portions of code would be accusable only in case compiler by Intel compilers. So non intel compilers likely would get lower perf comparing to version build with Intel compiler. 

But overall would see if ClangCL can be added

**Comment by emmenlau** [comment]
@napetrov what you say sounds very reasonable! Would love to have ClangCl added (even with these limitations).

**Comment by david-cortes-intel** [comment]
It is now possible to build the library with clang through the `COMPILER` option.

@icfaust Could you please add instructions for doing so in the `INSTALL.md` file?

**Comment by icfaust** [comment]
Sorry @david-cortes-intel I do not claim ownership of our makefile nor the ability to compile with various compilers.  I recommend adding this to the backlog and it will be assigned as the manpower budget allows for it.

**Comment by nithin-aashik-mcw** [comment]
@emmenlau If you have changed make files to support clang-cl, then add give flags to make it work.
>-Werror -Wno-empty-body -Wreturn-type -Wno-deprecated-declarations

I'm not sure this will work. But you can try.
